src/utils.ts

```diff
@@ -1,89 +1,124 @@
- * @param contractAddress - address of the contract to check 
- * @param signatureHash - keccak256 of the method signature
- * 
- * @returns
- * true if method exists, false if not
- */
-export const hasMethod = async(web3: Web3, contractAddress: string, signatureHash: string) => {
-  const code = await web3.eth.getCode(contractAddress);
-  return code.indexOf(signatureHash.slice(2, signatureHash.length)) > 0;
-}
-
-/**
- * Checks if the given web3 instance has accounts to sign txs
- * 
- * @param web3 - Web3 instance 
- * 
- * @returns
- * true if has accounts, false if not
- */
-export const hasAccounts = async(web3: Web3) => {
-  let accounts = [];
-  try {
-    accounts = await web3.eth.getAccounts();
-  } catch {
-    return false;
+export default class implements Utils {
+  
+  constructor() { }
+  
+  /**
+   * Checks if the contract in the given address has the given method 
+   * 
+   * @param web3 - Web3 instance 
+   * @param contractAddress - address of the contract to check 
+   * @param signatureHash - keccak256 of the method signature
+   * 
+   * @returns
+   * true if method exists, false if not
+   */
+  async hasMethod(web3: Web3, contractAddress: string, signatureHash: string): Promise<boolean>{
+    const code = await web3.eth.getCode(contractAddress);
+    return code.indexOf(signatureHash.slice(2, signatureHash.length)) > 0;
   }
-  return accounts.length > 0;
-}
 
-/**
- * Validates the given label
- * 
- * @param label - label to validate
- * 
- * @returns
- * true if valid, false if not
- */
-export const validLabel = (label: string) => {
-  return label && !label.match('[^a-z0-9]');
-}
-
-/**
- * Validates the given domain syntax
- * 
- * @param domain - domain to validate
- * 
- * @returns
- * true if valid, false if not
- */
-export const validDomain = (domain:string) => {
-  const labels = domain.split('.');
+  /**
+   * Checks if the given web3 instance has accounts to sign txs
+   * 
+   * @param web3 - Web3 instance 
+   * 
+   * @returns
+   * true if has accounts, false if not
+   */
+  async hasAccounts(web3: Web3): Promise<boolean> {
+    let accounts = [];
+    try {
+      accounts = await web3.eth.getAccounts();
+    } catch {
+      return false;
+    }
+    return accounts.length > 0;
+  }
 
-  if (labels.length < 1) {
+  /**
+   * Validates the given label
+   * 
+   * @param label - label to validate
+   * 
+   * @returns
+   * true if valid, false if not
+   */
+  validLabel(label: string): boolean {
+    if (label && !label.match('[^a-z0-9]')) {
+      return true;
+    }
     return false;
   }
 
-  for (let i = 0; i < labels.length; i++) {
-    const label = labels[i];
+  /**
+   * Validates the given domain syntax
+   * 
+   * @param domain - domain to validate
+   * 
+   * @returns
+   * true if valid, false if not
+   */
+  validDomain(domain:string): boolean {
+    const labels = domain.split('.');
 
-    if (!validLabel(label)) {
+    if (labels.length < 1) {
       return false;
     }
+
```

Comment:
`isValidTld(labels[labels.length - 1]`?

Reply:
Doesn't make sense to place that validation here, because we want to validate just the syntax of the domain. The syntax does not care on TLDs